app/models/alchemy/picture_extension.rb

```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 Alchemy::Picture.module_exec do
+  has_many :dragonfly_signatures, dependent: :destroy
```

Comment:
Shouldn't the picture just destroy its versions, which in turn should destroy their signatures? It seems to me that only either of the picture or the version should be responsible for destroying the signatures, not both.

Reply:
The problem is that while there is a picture_id field on picture_versions, it's always empty for any versions in our db, also there's no code to connect pictures to picture_versions apart from the dragonfly signature.
Since the dragonfly signatures have been sort of retrofitted in order to connect the versions and pictures, this is a bit messy. 
I've added some comments and revised the association declarations, to hopefully make this a bit more transparent.